idseq_dag/util/s3.py

```diff
@@ -140,9 +152,22 @@ def fetch_from_s3(src,  # pylint: disable=dangerous-default-value
         # This check is a bit imperfect when untarring... unless you follow the discipline that
         # all contents of file foo.tar are under directory foo/...
         if os.path.exists(dst):
-            # No need to fetch this file from s3, it has been just produced
-            # on this instance.
-            return dst
+            # Destination filename exists. If it is in the reference download directory, check the reference fetch log
+            # for a record of the source key and etag, and short-circuit the download if they match. Otherwise, blindly
+            # assume that it's the same file and short-circuit the download.
+            if abspath.startswith(config["REF_DIR"]):
+                try:
+                    with open(os.path.join(config["REF_FETCH_LOG_DIR"], abspath_hash)) as fh:
```

Comment:
Considering IDseq uses `multiprocessing` package and have tons of separate processes, should we protect the code section that is reading / writing json files with a multiprocessing Lock?

Reply:
No, because this section is protected by the lock on line 139/151 (https://github.com/chanzuckerberg/idseq-dag/pull/218/files#diff-30956c749673657f32d6e09a0ac2e49dR151).